File: java/abc175/C.java
import java.util.*;

public class C {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    long x = sc.nextLong();
    long k = sc.nextLong();
    long d = sc.nextLong();

    long abs_x = Math.abs(x);
    long moved_distance = abs_x - (abs_x % d);
    long moved_count = moved_distance / d;

    if (moved_count >= k) {
      System.out.println(abs_x - k * d);
      return;
    }

    if ((k - moved_count) % 2 == 0) {
      System.out.println(abs_x - moved_distance);
      return;
    }

    System.out.println(Math.abs(abs_x - moved_distance - d));

  }
}

File: rust/abc175/src/c.rs
use proconio::input;
use std::io::{stdout, BufWriter, Write};

pub fn main() {
  let out = stdout();
  let mut out = BufWriter::new(out.lock());
  input! {x: isize, k: usize, d: usize};

  let abs_x: usize = x.abs() as usize;
  let moved_distance = abs_x - abs_x % d;
  let moved_times = moved_distance / d;
  if moved_times >= k {
    writeln!(out, "{}", abs_x - k * d).unwrap();
    return;
  }
  if (k - moved_times) % 2 == 0 {
    writeln!(out, "{}", abs_x - moved_distance).unwrap();
    return;
  }
  writeln!(
    out,
    "{}",
    ((abs_x - moved_distance) as isize - d as isize).abs()
  )
  .unwrap();
}
